### PR TITLE
Fixes Wheelchairs and other vehicles transcending space and time.

### PR DESCRIPTION
**Pull request by BurgerLUA**
Basically ports https://github.com/Baystation12/Baystation12/pull/28003

If you try to move up or down in your vehicle, you'll just get off.

### CODE REVIEW

**Comment by Sparrowbird611** [comment]
You can't tell me that interdimensional wheel chairs aren't a feature.

Jokes aside, this is good, considering the fact that you could literally phase through walls and people with these things.

**Comment by Roland410** [comment]
Won't this mean you also get unbuckled every time you move up or down with the elevator or stairs? Also, does the spacebike retain its proper functionality, considering you use those to fly in space and between decks?

**Comment by MarinaGryphon** [comment]
> Won't this mean you also get unbuckled every time you move up or down with the elevator or stairs?

No, that uses `Move()`, not `relaymove()`.



> Also, does the spacebike retain its proper functionality, considering you use those to fly in space and between decks?

The spacebike will probably be broken by this fix as-is, so that needs to be addressed, yes.

File: code/modules/vehicles/bike.dm [hunk 1]
```diff
@@ -133,6 +133,11 @@
 		to_chat(user, "You unbuckle yourself from \the [src]")
 
 /obj/vehicle/bike/relaymove(mob/user, direction)
+
+	if(direction & (UP|DOWN))
```
Comment (MarinaGryphon) [inline code/modules/vehicles/bike.dm]: Space bikes are supposed to be able to move up and down in space. Instead, I believe there are some other checks that would be applicable here.

Comment (MarinaGryphon) [inline code/modules/vehicles/bike.dm]:
```suggestion
	if(direction & (UP|DOWN) && !isopenspace(get_step(user, direction)))
```
Apparently we don't have an ACTUAL multiz adjacency/reachability check. Huh.

Comment (MarinaGryphon) [reply code/modules/vehicles/bike.dm]: Nvm. Am big dumb.
```suggestion
	if(direction & (UP|DOWN) && !CanMoveOnto(src, get_step(src, direction))
```

Comment (MarinaGryphon) [reply code/modules/vehicles/bike.dm]: Nvm. Am bigger dumb. Don't use that one.

Comment (MarinaGryphon) [reply code/modules/vehicles/bike.dm]: May also cause issues with nonconnected Z-levels suddenly becoming connected if there's an openspace on the bottom or top level. That would be bad. This PR is fine as a stopgap fix.

Comment (BurgerLUA) [reply code/modules/vehicles/bike.dm]: ok so which one do you want me to use

Comment (MarinaGryphon) [reply code/modules/vehicles/bike.dm]: Neither, use the one you have.

**Comment by MemedHams** [comment]
duplicate the current wheelchair before fixing it and rename the copy "Bluespace Wheelchair", this is too excellent a capability to not be preseved

**Pull request merged**